sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,258 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+          .getRepository(EducationProgramOffering)
+          .save(offering);
+
+        // Create the new assessment to be processed.
+        scholasticStanding.studentAssessment = {
+          application: { id: application.id } as Application,
+          triggerType: AssessmentTriggerType.ScholasticStandingChange,
+          creator: auditUser,
+          submittedBy: auditUser,
+          submittedDate: now,
+          offering: { id: savedOffering.id } as EducationProgramOffering,
+        } as StudentAssessment;
+      } else {
+        // If unsuccessful weeks, then add to the column.
+        // * No cloning of offering and re-assessment is required in this scenario.
+        scholasticStanding.unsuccessfulWeeks =
+          scholasticStandingData.numberOfUnsuccessfulWeeks;
+      }
 
       const studentScholasticStanding = await transactionalEntityManager
         .getRepository(StudentScholasticStanding)
         .save(scholasticStanding);
 
-      // Save current application.
-      application.currentAssessment = {
-        id: scholasticStanding.studentAssessment.id,
-      } as StudentAssessment;
+      if (scholasticStanding.studentAssessment) {
+        // Save current assessment to application, if any.
+        application.currentAssessment = {
+          id: scholasticStanding.studentAssessment.id,
+        } as StudentAssessment;
+      }
 
       // Set archive to true
       application.isArchived = true;
 
+      // Save current application.
       await transactionalEntityManager
         .getRepository(Application)
         .save(application);
 
       return studentScholasticStanding;
     });
   }
+
+  /**
+   * Process the payload data and checks for certain restriction,
+   * and add new restrictions, if required.
+   * When institution report Withdrawal OR unsuccessful for a PT course application,
+   * add PTSSR restriction to student.
+   * If a ministry user resolves the SSR or PTSSR or WTHD restriction, and new withdrawal
+   * is reported, re add the above restrictions.
+   * @param scholasticStandingData scholastic standing data.
+   * @param offeringIntensity offering intensity.
+   * @param studentId student id.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   * @param applicationId application id.
+   * @returns a new student restriction object, that need to be saved.
+   */
+  async getStudentRestrictions(
+    scholasticStandingData: ScholasticStanding,
+    offeringIntensity: OfferingIntensity,
+    studentId: number,
+    auditUserId: number,
+    applicationId: number,
+  ): Promise<StudentRestriction> {
+    if (offeringIntensity === OfferingIntensity.fullTime) {
+      return this.getFullTimeStudentRestrictions(
+        scholasticStandingData,
+        studentId,
+        auditUserId,
+        applicationId,
+      );
+    }
+    if (offeringIntensity === OfferingIntensity.partTime) {
```

Comment:
Apparently, not all paths return a value in this method. Can we please add a `return null` in the end and make the return type explicit for this one `Promise<StudentRestriction | null>`?